packages/chain-adapters/src/solana/SolanaChainAdapter.ts

```diff
@@ -380,32 +384,40 @@ export class ChainAdapter implements IChainAdapter<KnownChainIds.SolanaMainnet>
   ): Promise<FeeDataEstimate<KnownChainIds.SolanaMainnet>> {
     try {
       const { baseFee, fast, average, slow } = await this.providers.http.getPriorityFees()
+      const { sendMax, chainSpecific } = input
+      const { instructions } = chainSpecific
 
       const serializedTx = await this.buildEstimationSerializedTx(input)
-      const computeUnits = await this.providers.http.estimateFees({
+      const baseComputeUnits = await this.providers.http.estimateFees({
         estimateFeesBody: { serializedTx },
       })
 
+      const computeUnits = sendMax
+        ? bnOrZero(baseComputeUnits).times(SOLANA_COMPUTE_UNITS_BUFFER_MULTIPLIER).toFixed()
+        : baseComputeUnits
+
+      const instructionNumber = instructions?.length || SOLANA_MINIMUM_INSTRUCTIONS_NUMBER
+
       return {
         fast: {
           txFee: bn(microLamportsToLamports(fast))
             .times(computeUnits)
-            .plus(baseFee)
-            .toFixed(0, BigNumber.ROUND_HALF_UP),
+            .plus(bnOrZero(baseFee).times(instructionNumber))
```

Comment:
This is very weird that it worked without this, it's mainly because solana was taking less fees than we thought, but big TXs can revert because of this

Reply:
Splitted to https://github.com/shapeshift/web/pull/8328